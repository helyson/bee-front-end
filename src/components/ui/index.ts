export { UIAvatar } from "./avatar/Avatar";
export { UIButton } from "./button/Button";
export { UIFormGroup } from "./form-group/FormGroup";
export { UIInput } from "./input/Input";
export { UILogo } from "./logo/Logo";
export { UIModal } from "./modal/Modal";
export { UISvgIcon } from "./icon/Icon";
export { UITable } from "./table/Table";
