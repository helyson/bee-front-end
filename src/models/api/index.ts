export { RequestEnum } from "./request.enum";
export * from "./status";
export * from "./request-api";
