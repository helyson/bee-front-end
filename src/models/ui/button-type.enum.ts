enum ButtonTypeEnum {
    Button = "btn",
    Link = "link",
    TextIcon= "textIcon"
  }
  
  export { ButtonTypeEnum};