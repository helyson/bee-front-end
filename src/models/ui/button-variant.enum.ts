enum ButtonVarianEnum {
  Primary = "primary",
  Secondary = "secondary",
  Gray = "gray",
}

export { ButtonVarianEnum };
