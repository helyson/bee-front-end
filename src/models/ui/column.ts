import { ReactNode } from "react";

interface Column<T> {
  key: string;
  label: string;
  width?: string
  render: (record: T, isHover?:boolean) => ReactNode;
}

export type { Column };
