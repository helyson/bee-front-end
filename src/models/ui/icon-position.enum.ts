enum IconPositionEnum {
  Left = "left",
  Right = "right",
}

export { IconPositionEnum };
