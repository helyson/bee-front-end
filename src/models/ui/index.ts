export { ButtonTypeEnum } from "./button-type.enum";
export { ButtonVarianEnum } from "./button-variant.enum";
export { IconEnum } from "./icon.enum";
export { IconPositionEnum } from "./icon-position.enum";
export * from "./column";
