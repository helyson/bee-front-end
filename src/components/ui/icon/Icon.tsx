import React, { CSSProperties } from "react";
import { getPath } from "./PathSvg";
import { getViewBox } from "./BoxSvg";
import { IconEnum } from "src/models/ui";

interface IconProps {
  name: IconEnum;
  style?: CSSProperties;
  fill?: string;
  viewBox?: string;
  width?: string;
  className?: string;
  height?: string;
}
const UISvgIcon: React.FC<IconProps> = ({
  name,
  style = {},
  fill = "",
  viewBox,
  width = "100%",
  className = "",
  height = "100%",
}) => {
  return (
    <>
      <svg
        data-testid="svg-element"
        width={width}
        style={style}
        height={height}
        className={className}
        viewBox={viewBox || getViewBox(name)}
      >
        {getPath(name, { fill })}
      </svg>
    </>
  );
};
export { UISvgIcon };