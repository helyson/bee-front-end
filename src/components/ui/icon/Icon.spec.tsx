import React from "react";
import { cleanup, render } from "@testing-library/react";
import { UISvgIcon } from "./Icon";
import { IconEnum } from "src/models/ui";

describe(" <UISvgIcon />", () => {
  afterEach(cleanup);
  test("should be render icon", () => {
    const { container } = render(
      <UISvgIcon
        name={IconEnum.Plus}
        className="beetrack-fill-primary"
        width="28"
        height="28"
      />
    );
    expect(container).toMatchSnapshot();
  });
  test("should be render icon basic", () => {
    const { container } = render(
      <UISvgIcon
        name={IconEnum.Plus}
      />
    );
    expect(container).toMatchSnapshot();
  });
});