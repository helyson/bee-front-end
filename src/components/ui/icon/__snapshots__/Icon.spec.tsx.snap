// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <UISvgIcon /> should be render icon 1`] = `
<div>
  <svg
    class="beetrack-fill-primary"
    data-testid="svg-element"
    height="28"
    viewBox="0 0 16 16"
    width="28"
  >
    <path
      d="M8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zM13 9h-4v4h-2v-4h-4v-2h4v-4h2v4h4v2z"
      fill=""
    />
  </svg>
</div>
`;

exports[` <UISvgIcon /> should be render icon basic 1`] = `
<div>
  <svg
    class=""
    data-testid="svg-element"
    height="100%"
    viewBox="0 0 16 16"
    width="100%"
  >
    <path
      d="M8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zM13 9h-4v4h-2v-4h-4v-2h4v-4h2v4h4v2z"
      fill=""
    />
  </svg>
</div>
`;
